src/app/clients/view-client/view-client.component.ts

```diff
@@ -87,6 +92,19 @@ export class ViewClientComponent implements OnInit, OnDestroy {
       );
   }
 
+  /**
+   * Fetches the image of the client if it exists.
+   * @param id Id of the client whose image is to be fetched
+   */
+  getClientImage(id: any) {
+    this.clientService.getClientImage(id)
```

Comment:
Use resolver here.

Reply:
This function will be called in the `getClientId` function, only if the image exists, so that one should be in the resolver. This can't be in the resolver because if the image doesn't exist and this function is called, 404 HTTP error is thrown and the app crashes. @luckyman20

Reply:
Oh, yeah, right. I forgot that resolver works asynchronously.

Reply:
Please add a comment:
```
// TODO: Shift into resolver when code will be refactored.
```